internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -65,29 +65,33 @@ func (nviu NodeVersionInfoUpdate) IsUpdated() bool {
 		nviu.Current.ContainerRuntimeVersion.Minor() == nviu.Update.ContainerRuntimeVersion.Minor()
 }
 
-func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded() bool {
+func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded() (bool, error) {
 	isControlPlane := nviu.Current.IsControlPlane()
-	currentClusterVersion, _ := kubeadm.GetCurrentClusterVersion()
+	client, err := kubernetes.GetAdminClientSet()
+	if err != nil {
+		return false, errors.Wrap(err, "unable to get admin client set")
+	}
+	currentClusterVersion, _ := kubeadm.GetCurrentClusterVersion(client)
 	allControlPlanesMatchVersion, _ := kubernetes.AllControlPlanesMatchVersion(currentClusterVersion)
```

Comment:
Now that we are at it, can we also retrieve the error in `GetCurrentClusterVersion` and `AllControlPlanesMatchVersion` as well?

Reply:
Fixed.

Reply:
Fixed.